### ISSUE TITLE
Enable running the Config UI as part of the Python Webserver

### ISSUE THREAD

**Issue by ThomasLaPiana** [ethyca/fides#560]
### Is your feature request related to a specific problem?

We need to enable the FastAPI webserver to serve the config UI without also running a node webserver

### Describe the solution you'd like

Export the next.js project as a static site that can be served by FastAPI

### Describe alternatives you've considered, if any

_A description of any alternative solutions or features you've considered._

### Additional context

_Add any other context or screenshots about the feature request here._

**Comment by ThomasLaPiana** [comment]
can also see github.com/ethyca/fidesops for inspiration

**Comment by allisonking** [comment]
It doesn't look like fidesops has implemented the serving part yet. From a quick google, FastAPI lets you [host static files pretty easily](https://fastapi.tiangolo.com/tutorial/static-files/). In the past, what I've done for this kind of problem is:

* Add a script in the frontend `package.json` along the lines of `copy-build` which 1. builds the static files and 2. copies them over to the proper place in the backend (where we'd point the FastAPI static file folder)
  * Your Dockerfile would also run `copy-build` 
  * You wouldn't commit the static files anywhere 
* If configured properly, the API server will then be able to host the static files at whatever port the server runs on (perhaps `8080`). However, you can still do a `npm run dev` to have your frontend dev environment at another port (maybe `3000`) for all your hot reloading dev needs. The static site at `8080` of course won't update unless you do another `copy-build`. In my experience this worked decently well, where frontend devs pretty much solely use `3000` and backend devs can use either port (depending on if they need a really up to date version of the frontend).

Would love to hear other approaches!

**Comment by ThomasLaPiana** [comment]
That seems like pretty much exactly what we want! In the `prod` stage of our Docker build we can build the static files and have FastAPI serve them. We can set up the docker-compose file to run `npm` for the local dev hot reloading then.

@NevilleS any objections to this set up? This fits with the flow I was imagining

**Comment by NevilleS** [comment]
I definitely think that'll work. In next, you can build the highly optimized prod version with `next build`, and it should actually be a completely static site. Technically this won't be possible yet though because of the auth library we're using (see https://github.com/ethyca/fidesui/issues/2). So, for the "production" Docker image we could build the static files and serve them up with uvicorn.

For bonus points, though, can we get it working in dev mode too in the same way? All we'd need to do is have uvicorn act as a reverse proxy to the node app... e.g. have the node app running on port 3000, and then have uvicorn just route all requests for those "static" assets to it

**Comment by ThomasLaPiana** [comment]
> I definitely think that'll work. In next, you can build the highly optimized prod version with `next build`, and it should actually be a completely static site. Technically this won't be possible yet though because of the auth library we're using (see [ethyca/fidesui#2](https://github.com/ethyca/fidesui/issues/2)). So, for the "production" Docker image we could build the static files and serve them up with uvicorn.
> 
> For bonus points, though, can we get it working in dev mode too in the same way? All we'd need to do is have uvicorn act as a reverse proxy to the node app... e.g. have the node app running on port 3000, and then have uvicorn just route all requests for those "static" assets to it

I would assume this is doable but have only really messed with uvicorn in the context of FastAPI

**Comment by ThomasLaPiana** [comment]
> The other thing we may need to worry about is backend paths vs frontend paths. i.e. right now it looks like there are API paths at `/policy`, `/organization`, etc. This can start to clash with frontend paths if we also want a page in the frontend at the same path (assuming we want the frontend to just live at localhost:8080). Ideally the api paths can be prefixed with i.e. `/api/policy`, but that might be a big documentation change?

@conceptualshark is here to save the day when we need to do those kinds of difficult docs changes! so that shouldn't stop us from going with the implementation that makes the most sense

also I think you're totally right, prepending all of the api paths with `/api` feels like the best route (pun intended) here

**Comment by ThomasLaPiana** [comment]
@allisonking can you open a separate ticket for migrating all of the endpoints to `/api`?

**Comment by allisonking** [comment]
I've got this mostly going [over here](https://github.com/ethyca/fides/pull/621), but am running into a problem I could use some guidance on. Right now I'm making a new directory in `fidesapi` called `build/static` where the frontend build can go. However, I don't want to commit these files, so I've added the path to the `.gitignore`. Locally I can populate that folder with a frontend and things get served up properly, but things get tricky in CI since those files don't exist in the git repo. 

I tried to have the dockerfile populate that folder with some temp data for now (since we don't have a frontend yet) but that appears to get overwritten by the docker compose volume mounting. i.e. if I comment out these few lines https://github.com/ethyca/fides/blob/aking-560-serve-static-files/docker-compose.yml/#L17-L21 then the tests will pass, but if I leave them in then there will be an error saying it can't find the `build/static` folder.

I believe this would still be a problem even once we have a full frontend since we wouldn't have those files in the repo, and from what I can tell the docker-compose will overwrite with just what is in the repo.
